### PR TITLE
fix dropout bug

### PR DESCRIPTION
**Pull request by lucienwang1009**
1. Refine dropout matching pattern.
2. **When delete unused nodes, we will re-add PlaceholderWithDefault to keep the graph complete, but ignore its input const. These const shouldn't be deleted either!!!**

### CODE REVIEW

**Comment by lucienwang1009** [comment]
Remove Dropout with ratio 1.0.

**Comment by nbcsm** [comment]
@mindest for awareness, please sync with @lucienwang1009 for details.

File: tests/run_pretrained_models.py [hunk 1]
```diff
@@ -242,6 +242,7 @@ def run_test(self, name, backend="caffe2", debug=False, onnx_file=None, opset=No
                 inputs[k] = v
 
         graph_def = tf2onnx.tfonnx.tf_optimize(inputs.keys(), self.output_names, graph_def, fold_const)
+        utils.save_protobuf(os.path.join(TEMP_DIR, name + "_after_tf_optimize.pb"), graph_def)
```
Comment (nbcsm) [inline tests/run_pretrained_models.py]: save only if it is debug mode?
it would be very slow to save large graph.

File: tests/test_backend.py [hunk 2]
```diff
@@ -353,6 +353,21 @@ def test_dropout(self):
...
+        x = tf.placeholder(tf.float32, shape=x_val.shape, name="input_1")
+        x_ = tf.identity(x)
+
+        fc1 = tf.nn.dropout(x_, keep_prob)
+
+        _ = tf.identity(fc1, name="output")
+        feed_dict = {"input_1:0": x_val}
+        input_names_with_port = ["input_1:0"]
+        output_names_with_port = ["output:0"]
+        self.run_test_case(feed_dict, input_names_with_port, output_names_with_port, constant_fold=False)
```
Comment (nbcsm) [inline tests/test_backend.py]: please add a few comments for constant_fold=False

File: tf2onnx/tfonnx.py [hunk 3]
```diff
@@ -216,6 +216,12 @@ def rewrite_dropout(g, ops):
         for n in set(match.get_nodes()):
             g.remove_node(n.name)
 
+    # remove dropout if its ratio is 1.0
+    for node in g.get_nodes():
```
Comment (nbcsm) [inline tf2onnx/tfonnx.py]: no need to go through all nodes, we can get all dropout node names easily.

Comment (lucienwang1009) [reply tf2onnx/tfonnx.py]: don't want to mix these two (actually) rewriter up.

Comment (nbcsm) [reply tf2onnx/tfonnx.py]: make sense.
